### PR TITLE
Refactor some codes and fixes some bugs.

### PR DESCRIPTION
**Pull request by bigstark**
## Refactor some codes
* create api / ui (main / location / info) and move to them.
* edit xml code in activity_main cause moving package. ( AppBarLayoutBehavor )

## Refactor some api codes.
* Remove multiple retrofit objects
* Remove unnecessary query params and replace it to endpoint.
* Concatenate getTMPosition / getStation / getAirInfo
* Remove null-safe code because return type is null safe.
* Add DataRepository for preparing local caching.

## Fixes some bugs.
* Remove notifyDataSetChanged in addData. It is called in iterator. notifyDataSetChanged must be called after interator finished.
* Add notifyDataSetChanged when get response in weather / forecast.

This PR would fix the bugs that won't be showing temperature / forecast / weather ...

### CODE REVIEW

File: app/src/main/java/com/mashupgroup/weatherbear/Global.kt [hunk 1]
```diff
@@ -28,14 +29,7 @@ object Global {
...
         // gson을 이용하여 addressList를 각각 json으로 직렬화한 후, 직렬화된 문자열을 저장한다
-        val gson = Gson()
-        val addrJsonArr = JSONArray()
-        for(address in addressList) {
-            val addrJson = gson.toJson(address)
-            addrJsonArr.put(addrJson)
-        }
-
-        editor.putString(SP_ADDR_SAVE_KEY, addrJsonArr.toString())
+        editor.putString(SP_ADDR_SAVE_KEY, Gson().toJson(addressList))
```
Comment (jujinkim) [inline app/src/main/java/com/mashupgroup/weatherbear/Global.kt]: 아 이렇게간단한방법이..

**Comment by jujinkim** [comment]
I've reviewed my parts. @YuChocopie  @pengin7384  Please review and learn some good things from @bigstark 's commit. There are so many awesome things to learn in his commit. These things would be great to help to improve your skills.

@bigstark Thanks for contributing our project, and we'll add your name into our information screen as a contributor if you don't mind.

This commit has conflicts so cannot be merged right now. I'll pull this commit and merge with resolve conflicts after our team review this.

File: app/src/main/java/com/mashupgroup/weatherbear/ui/main/MainPagerAdapter.kt [hunk 2]
```diff
@@ -26,11 +21,8 @@ class MainPagerAdapter : PagerAdapter {
     }
 
     override fun instantiateItem(container: ViewGroup, position: Int): Any {
-        val layoutInflater = context.getSystemService(Context.LAYOUT_INFLATER_SERVICE)
-                as LayoutInflater
-
-
-        val binding: ItemWeatherBinding = DataBindingUtil.inflate(layoutInflater, R.layout.item_weather, container, false)
+        val binding: ItemWeatherBinding = DataBindingUtil.inflate(LayoutInflater.from(container.context),
```
Comment (jujinkim) [inline app/src/main/java/com/mashupgroup/weatherbear/ui/main/MainPagerAdapter.kt]: 두 동작이 동일한가요? 다르다면, 차이점 설명해주실 수 있나요?
(LayoutInflator.from(container.context) 와 Activity.context.getSystemService(Context.LAYOUT_INFLATER_SERVICE)의 차이)

**Comment by YuChocopie** [comment]
@bigstark 프로젝트의 코드를 멋지게 만들어주셔서 정말 감사합니다 :)
현재 충돌로 merge 안되어서 rebase 후 커밋해주신다면 출시 버전 업데이트 해서 반영하겠습니다!
코드가 너무 예뻐져서 코틀린이 너무 사랑스러운 언어인걸 깨닫게 됐네요! 새로운거 많이 배워 갑니다!ㅎㅎ 👍

**Pull request merged**